addons/account/models/account.py

```diff
@@ -612,7 +612,8 @@ def onchange_price_include(self):
     def get_grouping_key(self, invoice_tax_val):
         """ Returns a string that will be used to group account.invoice.tax sharing the same properties"""
         self.ensure_one()
-        return str(invoice_tax_val['tax_id']) + '-' + str(invoice_tax_val['account_id']) + '-' + str(invoice_tax_val['account_analytic_id'])
+        return str(invoice_tax_val['tax_id']) + '-' + str(invoice_tax_val['account_id']) + '-' + str(invoice_tax_val['account_analytic_id'] or 0) \
```

Comment:
@wtaferner I wasn't 100% sure, but it seemed to me that after your refactoring of the base amount computed `invoice_tax_val['account_analytic_id']` could stringify `None` or `False` according to where it's called from (`... or False` vs `account_analytic_id.id`), resulting in an undesired hash here... I wasn't sure, but I hope this does no harm either.